### PR TITLE
GA event tracking

### PR DESCRIPTION
**Pull request by kewitham**
[Issue](https://github.com/usdoj-crt/crt-portal-management/issues/1548)

## What does this change?
This PR updates the GA tag on the site to GA4 and adds tracking for the following events:
- Search filtering
- Dashboard filtering
- Grouping clicks
- Related report clicks

Note that this is a draft PR subject to further discussion and the events added here have not been added in the GA dashboard yet.

### Author

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [ ] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/templates/forms/complaint_view/index/grouping-select.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <select name="grouping" class="usa-select select-link usa-button--unstyled" aria-label="Grouping options">
   <option value="default">Default</option>
-  <option value="matching-descriptions">Matching descriptions</option>
+  <option onselect="gtag('event', 'click', {'event_name': 'group by matching description'});"  value="matching-descriptions">Matching descriptions</option>
```
Comment (okkays) [inline crt_portal/cts_forms/templates/forms/complaint_view/index/grouping-select.html:3]: It'd be neat to capture section info with these.

We don't have a way to see what section the current user is from, but we could pass along the selected section filter?

File: crt_portal/cts_forms/templates/base.html [hunk 2]
```diff
@@ -161,6 +161,7 @@ <h1 class="usa-alert__heading">Test site</h1>
        function gtag(){dataLayer.push(arguments);}
        gtag('js', new Date());
        gtag('config', 'UA-176942176-1');
+       gtag('config', 'G-2T865GKYYK');
```
Comment (okkays) [inline crt_portal/cts_forms/templates/base.html:164]: 💯

File: crt_portal/cts_forms/templates/forms/complaint_view/show/description.html [hunk 3]
```diff
@@ -9,6 +9,7 @@ <h3 class="complaint-card-heading text-uppercase">
       Personal description
       <small class="margin-left-auto">
         <a class="related-reports"
+           onclick="gtag('event', 'click', {'event_name': 'view related reports'});"
```
Comment (okkays) [inline crt_portal/cts_forms/templates/forms/complaint_view/show/description.html:12]: how might we add this to components generated by django (a sufficiently annoying example might be the DJ number filter?)

File: crt_portal/cts_forms/templates/forms/complaint_view/dashboard/filter-controls.html [hunk 4]
```diff
@@ -13,7 +13,7 @@ <h2 class="intake-section-title">Team Management</h2>
     </div>
 
     <div class="text-right">
-      <button class="usa-button" type="submit" id="apply-filters-button">
+      <button class="usa-button" type="submit" id="apply-filters-button" onclick="gtag('event', 'click', {'event_name': 'dashboard filter'});">
```
Comment (okkays) [inline crt_portal/cts_forms/templates/forms/complaint_view/dashboard/filter-controls.html:16]: Is it possible to confirm that these work in local/dev/staging?

Either a separate GA property, or something simple like rigging gtag to console.log the events and params we're sending along?

**Review by okkays — COMMENTED** [review]
added some thoughts for discussion - thanks for taking a pass at this!

File: crt_portal/static/js/complaint_view_filters.js [hunk 5]
```diff
@@ -103,6 +103,7 @@
    * @returns {String} Param in the format of {key}={value}
    */
   function makeQueryParam(key, value) {
+    gtag('event', 'search_filter', { filter: key });
```
Comment (okkays) [inline crt_portal/static/js/complaint_view_filters.js:106]: is this going to make separate events for each param? if so, is that what we want?

This is kind of open for discussion I guess, but if you want to see what combinations are being used, maybe:

`[...Object.keys(params)].sort().join(' ')` to get a list of keys?

**Comment by ilodidoj** [comment]
@kewitham this branch has some CI/CD E2E test error, please check.

File: crt_portal/cts_forms/templates/forms/complaint_view/show/description.html (hunk 3 above)
Comment (kewitham) [reply crt_portal/cts_forms/templates/forms/complaint_view/show/description.html]: updated to use gaUtil JS file and track clicks by class names and ids

**Comment by kewitham** [comment]
> @kewitham please check the conflict in this branch and resolve it. Thanks.

@ilodidoj resolved!

**Pull request merged**